0004-remove-spandata.md

```diff
@@ -0,0 +1,43 @@
+# Remove SpanData
+
+Remove and replace SpanData by adding span start and finish options.
+
+## Motivation
+
+SpanData represents an immutable span object, creating a fairly large API for all of the fields (12 to be exact). It exposes what feels like an SDK concern and implementation detail to the API surface. As a user, this is another API I need to learn how to use. As an implementer, it’s a new data type that needs to be supported. The primary motivation for removing SpanData revolves around the desire to reduce the size of the tracing API.
+
+## Explanation
+
+SpanData's primary use case comes from the need to construct and report out of band spans, meaning that you're creating "custom" spans for an operation you don't own. A good example of this is a program that takes in structured logs that contain correlation IDs and a duration (e.g. from splunk) and [converts them](https://github.com/lightstep/splunktospan/blob/master/splunktospan/span.py#L43) to spans for your tracing system. [Another example](https://github.com/lightstep/haproxy_log2span/blob/master/lib/lib.go#L292) is running a sidecar on an HAProxy machine, tailing the request logs, and creating spans. SpanData supports the out of band reporting case, whereas you can’t with the current Span API as you cannot set the start and end timestamp, or add any tags at span creation that the sampler might need.
+
+I'd like to propose getting rid of SpanData and `tracer.recordSpanData()` and replacing it by allowing `tracer.startSpan()` and `span.finish()` to accept start and end timestamp options. This reduces the API surface, consolidating on a single span type. Options would meet the requirements for out of band reporting.
```

Comment:
Maybe it was my misunderstanding, I thought `SpanData` was targeting consumers (e.g. exporters) where immutable is desired.
`Span` was targeting producers, where different components can grab the "current active span" from the context/tracer and append extra information.

Reply:
SpanData in the API has the motivation explained by the RFC. In the SDK may have a different usage, but if we need something in the SDK then we can define it there.

Reply:
There has been additional confusion because the exporters declare a `SpanData` type, but it's not the same thing that's been discussed as a way to handle "out of band" spans. This RFC is about removing `SpanData` from the API, not the observer.